subprojects/diagnostics/src/main/java/org/gradle/api/plugins/HelpTasksPlugin.java

```diff
@@ -180,10 +181,12 @@ public void execute(final DependencyInsightReportTask task) {
             task.setGroup(HELP_GROUP);
             task.setImpliesSubProjects(true);
             task.getShowingAllVariants().convention(false);
-            ComponentRegistry componentRegistry = ((ProjectInternal) task.getProject()).getServices().get(ComponentRegistry.class);
+            SoftwareComponentContainerInternal components = (SoftwareComponentContainerInternal) task.getProject().getComponents();
```

Comment:
As a few alternative ideas to this:

- We could move this configuration to Java plugin, since that's the only way that this will do anything.
- Or we could consider deprecating this behavior and require that someone passes `--configuration` all the time. 
- Or we could just look for a configuration named `compileClasspath` vs looking at the component.

I don't feel strongly about this, but it avoids having to name/create a "DiagnosableSoftwareComponent"

Reply:
I don't like how I needed to do this so I'm definitely open to ideas here. This is actually already causing me problems. 

Given your options: 
1) This could work, and I'll probably go with this for now, but given that the JavaPlugin is going away, I'm not sure thats the best place for this. Perhaps in the java base plugin? 
2) This would make everything easier for us. Do you think this would be heavy-handed? otherwise I would support this option
3) This is kinda going in the opposite direction than I am intending. I would not want to do this option

Reply:
1. It would live in the plugin that creates the component, so right now that's JavaPlugin
2. I think this is probably better for other reasons too. e.g., I think current behavior means that you get `compileClasspath` in any project that applies Java, even if the project isn't a "Java project" (e.g., KMP, AGP). And if we decided to change all of the diagnostic tasks to be "component centric", this convention would get in the way. 
3. I only offered this because it's effectively what's happening now with fewer steps.  Something unrelated to Java is looking around for something created by the Java plugin by name and grabbing it.  Instead of looking for the component, look for the configuration.